desc/objectives/_wrappers.py

```diff
@@ -89,7 +89,29 @@ def build(self, eq, use_jit=None, verbose=1):
             self._scalar = False
 
         # set_state_vector
-        self._args = ["p_l", "i_l", "c_l", "Psi", "Rb_lmn", "Zb_lmn"]
+
+        # this is everything taken by either objective
+        self._full_args = self._eq_objective.args + self._objective.args
+        self._full_args = [arg for arg in arg_order if arg in self._full_args]
+        # remove constraints that aren't necessary
+        self._constraints = tuple(
+            [con for con in self._constraints if con.args[0] in self._eq_objective.args]
+        )
+
+        # arguments being optimized are all args, but with internal degrees of freedom
+        # replaced by boundary terms
+        self._args = self._full_args.copy()
+        if "L_lmn" in self._args:
+            self._args.remove("L_lmn")
+        if "R_lmn" in self._args:
+            self._args.remove("R_lmn")
+            if "Rb_lmn" not in self._args:
+                self._args.append("Rb_lmn")
+        if "Z_lmn" in self._args:
+            self._args.remove("Z_lmn")
+            if "Zb_lmn" not in self._args:
+                self._args.append("Zb_lmn")
+        # and remove pressure if we're solving a vacuum equilibrium
         if isinstance(self._eq_objective.objectives[0], CurrentDensity):
             self._args.remove("p_l")
```

Comment:
Are these lines still necessary? If we are solving a vacuum equilibrium I think the list of args will already exclude pressure.

Reply:
Yeah I think we can get rid of it.